firedrake/preconditioners/hypre_ads.py

```diff
@@ -25,6 +25,8 @@ def initialize(self, obj):
             degree = max(degree)
         except TypeError:
             pass
+        if mesh.topological_dimension() != 3:
+            raise ValueError("Hypre ADS only works for meshes of topological dimension 3, (not degree %d)" % (mesh.topological_dimension())) 
```

Comment:
Should say dimension instead of degree. Also we should suggest "Did you mean HypreAMS?"